src/services/healthcareProfessionalService.ts

```diff
@@ -0,0 +1,1033 @@
+        const errorMessage = (error as Error).message
+
+        // Handle HP_NOT_FOUND error
+        if (errorMessage === 'HP_NOT_FOUND') {
+            logger.warn(`deleteHealthcareProfessional: professional not found: ${id}`)
+            return {
+                data: { isSuccessful: false },
+                hasErrors: true,
+                errors: [{
+                    field: 'deleteHealthcareProfessional',
+                    errorCode: ErrorCode.INVALID_ID,
+                    httpStatus: 404
+                }]
+            }
+        }
+
+        // Generic error
+        logger.error(`ERROR: Error deleting professional ${id}: ${error}`)
+        return {
+            data: { isSuccessful: false },
+            hasErrors: true,
+            errors: [{
+                field: 'deleteHealthcareProfessional',
+                errorCode: ErrorCode.INTERNAL_SERVER_ERROR,
+                httpStatus: 500
+            }]
+        }
+    }
+}
+
+/**
+ * Bulk updates HP-Facility relationships when a facility is modified.
+ * USE CASE:
+ * When a Facility is updated and its healthcareProfessionalIds change,
+ * this function updates the junction table (hps_facilities) accordingly.
+ * BUSINESS RULES:
+ * - HP must always be linked to at least ONE facility
+ * - Cannot delete a relation if it's the HP's last facility
+ * - Batch validation: checks all HPs before making any changes
+ * NOTE: No audit log in this function because it's called FROM updateFacility,
+ * which already creates an audit log for the facility change.
+ * 
+ * @param professionalRelationshipsToUpdate - Array of relationship changes (CREATE/DELETE)
+ * @param facilityId - The facility ID being linked/unlinked
+ * @returns Result indicating success or validation errors
+ */
+export async function updateHealthcareProfessionalsWithFacilityIdChanges(
+    professionalRelationshipsToUpdate: gqlTypes.Relationship[],
+    facilityId: string
+): Promise<Result<void>> {
+    try {
+        // Early return if no changes requested
+        if (!professionalRelationshipsToUpdate || professionalRelationshipsToUpdate.length < 1) {
+            return {
+                data: undefined,
+                hasErrors: false
+            }
+        }
+
+        // Execute all operations in atomic transaction
+        await db.transaction().execute(async trx => {
+            // Split relationships into CREATE vs DELETE
+            const hpIdsToCreate: string[] = professionalRelationshipsToUpdate
+                .filter(r => r.action === gqlTypes.RelationshipAction.Create)
+                .map(r => r.otherEntityId)
+
+            const hpIdsToDelete: string[] = professionalRelationshipsToUpdate
+                .filter(r => r.action === gqlTypes.RelationshipAction.Delete)
+                .map(r => r.otherEntityId)
+
+            // Resolve conflicts, if HP appears in both CREATE and DELETE, it's a no-op
+            const createSet = new Set<string>(hpIdsToCreate)
```

Comment:
we don't have a business case where it would have both, so I think we can remove this logic.